packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -363,35 +411,94 @@ class FlutterPlugin implements Plugin<Project> {
                 }
             }
 
-            FlutterTask flutterTask = project.tasks.create(name: "${flutterBuildPrefix}${variant.name.capitalize()}", type: FlutterTask) {
-                flutterRoot this.flutterRoot
-                flutterExecutable this.flutterExecutable
-                buildMode flutterBuildMode
-                localEngine this.localEngine
-                localEngineSrcPath this.localEngineSrcPath
-                targetPath target
-                verbose verboseValue
-                fileSystemRoots fileSystemRootsValue
-                fileSystemScheme fileSystemSchemeValue
-                trackWidgetCreation trackWidgetCreationValue
-                compilationTraceFilePath compilationTraceFilePathValue
-                createPatch createPatchValue
-                buildNumber buildNumberValue
-                baselineDir baselineDirValue
-                buildSharedLibrary buildSharedLibraryValue
-                targetPlatform targetPlatformValue
-                sourceDir project.file(project.flutter.source)
-                intermediateDir project.file("${project.buildDir}/${AndroidProject.FD_INTERMEDIATES}/flutter/${variant.name}")
-                extraFrontEndOptions extraFrontEndOptionsValue
-                extraGenSnapshotOptions extraGenSnapshotOptionsValue
+            def flutterTasks = []
+            def targetPlatforms = []
+
+            if (targetPlatformValue == null) {
+                if (flutterBuildMode == 'release') {
+                    targetPlatforms.addAll(allArmPlatforms)
+                } else {
+                    targetPlatforms.add('android-arm')
+                }
+            } else {
+                targetPlatforms.add(targetPlatformValue)
+            }
+
+            targetPlatforms.each { currentTargetPlatformValue ->
```

Comment:
nit: I think this and the nested lambdas in it would all be easier to follow if they were split out into named functions.

Reply:
`addFlutterJarApiDependency` needs some work. The closure created by `addFlutterDeps` makes the code a little more difficult to follow. In the meanwhile, I created https://github.com/flutter/flutter/issues/33064